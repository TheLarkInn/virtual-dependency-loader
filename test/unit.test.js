import fs from 'fs';
import { runLoaders } from 'loader-runner';
import Promise from 'bluebird';
import { getFixtureResource, getFixture, getLoader } from './test-utils';

const runLoadersPromise = Promise.promisify(runLoaders);
const readFilePromise = Promise.promisify(fs.readFile, { context: fs });


describe('inline-loader Tests: Fixture: simple-file', () => {
  const fixtureName = 'simple-file';
  const resource = getFixture(fixtureName);

  const mockReplacementCodeString = 'console.log("I am the real code")';
  const mockReplacementFilename = 'filename.imnotreal.js';

  const loaders = getLoader({
    code: mockReplacementCodeString,
    filename: mockReplacementFilename,
  });

  test('loaded file should be different', async () => {
    const originalSource = await readFilePromise(resource);
    const { result } = await runLoadersPromise({ resource: getFixtureResource(fixtureName), loaders });

    expect(result).not.toEqual(originalSource);
  });

  test('source of loader should be equal to code option', async () => {
    const result = await runLoadersPromise({ resource: getFixtureResource(fixtureName), loaders });
    const resultMatcher = expect.arrayContaining([
      expect.stringContaining('console.log("I am the real code")'),
    ]);

    expect(result.result).toEqual(resultMatcher);
    expect(result).toMatchSnapshot();
  });
});


describe('inline-loader Tests: Fixture: simple-file', () => {
  const fixtureName = 'simple-file';
  const resource = getFixture(fixtureName);

  const mockReplacementCodeString = `
    coonsole.log("!@!#@#@#%^$%#3wert34523452345fgfsdgsdfgsdfgasdfasdf".length);
  `;
  const mockReplacementFilename = 'filename.imnotreal.js';

  const loaders = getLoader({
    code: mockReplacementCodeString,
    filename: mockReplacementFilename,
  });

  test('loaded file should be different', async () => {
    const originalSource = await readFilePromise(resource);
    const { result } = await runLoadersPromise({ resource: getFixtureResource(fixtureName), loaders });

    expect(result).not.toEqual(originalSource);
  });

  test('source of loader should be equal to code option', async () => {
    const result = await runLoadersPromise({ resource: getFixtureResource(fixtureName), loaders });
    const resultMatcher = expect.arrayContaining([
      expect.stringContaining(mockReplacementCodeString),
    ]);

    expect(result.result).toEqual(resultMatcher);
    expect(result).toMatchSnapshot();
  });
});
