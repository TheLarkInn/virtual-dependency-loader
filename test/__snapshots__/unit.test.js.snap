// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example Loader Tests: Fixture: simple-file source of loader should be equal to code option 1`] = `
Object {
  "cacheable": true,
  "contextDependencies": Array [],
  "fileDependencies": Array [
    "/Users/wecomm/seanlarkin/code/virtual-dependency-loader/test/fixtures/simple-file.js",
  ],
  "resourceBuffer": Object {
    "data": Array [
      105,
      109,
      112,
      111,
      114,
      116,
      32,
      102,
      111,
      111,
      32,
      102,
      114,
      111,
      109,
      32,
      34,
      46,
      47,
      102,
      111,
      111,
      34,
      59,
      32,
      47,
      47,
      32,
      101,
      115,
      108,
      105,
      110,
      116,
      45,
      100,
      105,
      115,
      97,
      98,
      108,
      101,
      45,
      108,
      105,
      110,
      101,
      10,
      10,
      99,
      111,
      110,
      115,
      111,
      108,
      101,
      46,
      108,
      111,
      103,
      40,
      102,
      111,
      111,
      41,
      59,
      10,
    ],
    "type": "Buffer",
  },
  "result": Array [
    "console.log(\\"I am the real code\\")",
  ],
}
`;
